zerver/tests/test_auth_backends.py

```diff
@@ -2859,6 +2861,37 @@ def test_update_user_avatar(self) -> None:
             hamlet = self.example_user('hamlet')
             self.assertEqual(hamlet.avatar_source, UserProfile.AVATAR_FROM_USER)
 
+    @use_s3_backend
+    def test_update_user_avatar_for_s3(self) -> None:
+        bucket = create_s3_buckets(settings.S3_AVATAR_BUCKET)[0]
+        test_image_data = open(get_test_image_file('img.png').name, 'rb').read()
+
+        self.mock_ldap.directory = {
+            'uid=hamlet,ou=users,dc=zulip,dc=com': {
+                'cn': ['King Hamlet', ],
+                'thumbnailPhoto': [test_image_data, ],
+            }
+        }
+        with self.settings(AUTH_LDAP_USER_ATTR_MAP={'full_name': 'cn',
+                                                    'avatar': 'thumbnailPhoto'}):
+            self.perform_ldap_sync(self.example_user('hamlet'))
+
+        hamlet = self.example_user('hamlet')
+        path_id = user_avatar_path(hamlet)
+        original_image_path_id = path_id + ".original"
+        medium_path_id = path_id + "-medium.png"
+
+        original_image_key = bucket.get_key(original_image_path_id)
+        medium_image_key = bucket.get_key(medium_path_id)
+
+        image_data = original_image_key.get_contents_as_string()
+        self.assertEqual(image_data, test_image_data)
+
+        from zerver.lib.upload import resize_avatar, MEDIUM_AVATAR_SIZE
```

Comment:
Please move the imports to the top of the file; I only put them here to save time while hacking to get something working.
(similarly with the below)

Reply:
Made this change.